cips/cip-blob-verified-signer.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Support authored blobs
+description: Introduce a new blob type that can be submitted whereby the signer address is included and verified.
+author: Callum Waters (@cmwaters)
+discussions-to: https://forum.celestia.org/t/cip-blobs-with-verified-author
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-05-22
+---
+
+## Abstract
+
+Introduce a new v2 blob type that can be submitted with the author of the blob. Validators verify that the author is correct, simplifying the loop for rollups that adopt a fork-choice rule that enshrines a specific sequencer.
+
+## Motivation
+
+A common fork-choice rule for rollups is to enshrine the sequencer. In this situation, full rollup nodes, pull all the blobs on one or more namespaces and verify their authenticity through the address (i.e. `celestia1fhalcne7syg....`) that paid for those blobs, the `signer`. Currently in Celestia, the `signer` field, is located in the `MsgPayForBlobs` (PFB) which is separated from the blobs itself. Thus, the current flow is as follows:
+
+- Retrieve all the PFBs in the `PFBNamespace`. Verify inclusion and then loop through them and find the PFBs that correspond to the namespaces that the rollup is subscribed to.
+- For the PFBs of the namespaces that the rollup is subscribed to, verify the `signer` matches the sequencer.
+- Use the share indexes of the `IndexWrapper` of the PFB to retrieve the blobs that match the PFB. Verify the blobs inclusion and finally process the blobs.
+  
+For rollups, using ZK, such as the case with Soverign, the flow is as follows:
+
+- Query all the blobs from the rollup namespace via RPC
+- For each blob, reconstruct the blob commitment.
+- Fetch the PFB namespace
+- Parse the PFB namespace and create a mapping from blob commitment -> PFB
+- (In zero-knowledge) Accept the list of all rollup blobs and the list of relevant PFBs as inputs
+- (In zero-knowledge) Verify that the claimed list of blobs matches the block header using the namespaced merkle tree
+- (In zero-knowledge) For each blob, find the PFB with the matching commitment and check that the sender is correct.
+- (In zero-knowledge) For each relevant PFB, check that the bytes provided match the namespaced merkle tree
```

Comment:
> (In zero-knowledge)

From my perspective, I think I know what is meant by "zero-knowledge", but it might be valuable to be more specific or use different wording. I've heard the term "in circuit" used before for constraints applied to the public and private inputs, do you think that applies here or perhaps something else?

Reply:
I was taking this from a slack thread. I can rephrase to using (in circuit) if you think that would be better

Reply:
what do you think is best here @evan-forbes and @cmwaters, "in circuit" or "in zero-knowledge"?

Reply:
Changed to in circuit (https://github.com/celestiaorg/CIPs/pull/147/commits/62e5ef1ac8ffb785e5777cd837439d24b8a201f6)